### PR TITLE
TST(string dtype): Fix xfails in test_block_internals.py

### PR DESCRIPTION
**Pull request by WillAyd**

### CODE REVIEW

File: pandas/tests/frame/test_block_internals.py [hunk 1]
```diff
@@ -162,21 +160,7 @@ def test_constructor_with_convert(self):
         )
         tm.assert_series_equal(result, expected)
 
-    @pytest.mark.xfail(using_string_dtype(), reason="TODO(infer_string)")
     def test_construction_with_mixed(self, float_string_frame, using_infer_string):
-        # test construction edge cases with mixed types
```
Comment (WillAyd) [inline pandas/tests/frame/test_block_internals.py:167]: I assume something got lost in refactor, but the code removed here is not actually being tested. Both the `result` and `expected` variables are getting shadowed before they are compared

File: pandas/tests/frame/test_block_internals.py [hunk 2]
```diff
@@ -196,13 +180,11 @@ def test_construction_with_mixed(self, float_string_frame, using_infer_string):
...
         tm.assert_series_equal(result, expected)
 
-    @pytest.mark.xfail(using_string_dtype(), reason="TODO(infer_string)")
     def test_construction_with_conversions(self):
         # convert from a numpy array of non-ns timedelta64; as of 2.0 this does
         #  *not* convert
         arr = np.array([1, 2, 3], dtype="timedelta64[s]")
-        df = DataFrame(index=range(3))
-        df["A"] = arr
+        df = DataFrame({"A": arr})
```
Comment (WillAyd) [inline pandas/tests/frame/test_block_internals.py:187]: The failures in this module go back to the discussion in https://github.com/pandas-dev/pandas/issues/60338

I don't think it is important for these tests to use that construction pattern

File: pandas/tests/frame/test_block_internals.py [hunk 3]
```diff
@@ -401,14 +383,16 @@ def test_update_inplace_sets_valid_block_values():
     assert isinstance(df._mgr.blocks[0].values, Categorical)
 
 
-@pytest.mark.xfail(using_string_dtype(), reason="TODO(infer_string)")
 def test_nonconsolidated_item_cache_take():
     # https://github.com/pandas-dev/pandas/issues/35521
 
     # create non-consolidated dataframe with object dtype columns
```
Comment (jbrockmendel) [inline pandas/tests/frame/test_block_internals.py:389]: Is this still non-consolidated?

File: pandas/tests/frame/test_block_internals.py [hunk 4]
```diff
@@ -162,21 +160,7 @@ def test_constructor_with_convert(self):
...
-        data = [
-            [datetime(2001, 1, 5), np.nan, datetime(2001, 1, 2)],
-            [datetime(2000, 1, 2), datetime(2000, 1, 3), datetime(2000, 1, 1)],
-        ]
-        df = DataFrame(data)
-
-        # check dtypes
-        result = df.dtypes
-        expected = Series({"datetime64[us]": 3})
-
```
Comment (jbrockmendel) [inline pandas/tests/frame/test_block_internals.py:179]: Would adding a tm.assertsimething pass here?

Comment (WillAyd) [reply pandas/tests/frame/test_block_internals.py:179]: No - adding `tm.assert_series_equal(result, expected)` doesn't work on main irrespective of the string data type work.

The data provided is 3 dimensional so I'm not sure why the expected Series is just one element, unless it meant to do list multiplication instead of creating a dictionary

Comment (WillAyd) [reply pandas/tests/frame/test_block_internals.py:179]: I'm happy to make that change if we want the list multiplication. I just can't figure out why its here in the first place though, so figured a good removal candidate

File: pandas/tests/frame/test_block_internals.py (hunk 3 above)
Comment (WillAyd) [reply pandas/tests/frame/test_block_internals.py:389]: Ah nice catch - let me fix that

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Manual backport -> https://github.com/pandas-dev/pandas/pull/60781
